test/views/behaviours/ViewBehaviours.scala

```diff
@@ -23,60 +23,82 @@ trait ViewBehaviours extends ViewSpecBase {
 
   val expectTimeoutDialog: Boolean = true
 
-  protected def normalPage(
-    view: HtmlFormat.Appendable,
-    messageKeyPrefix: String,
-    messageKeySuffix: String,
-    messageHeadingArgs: Any*
-  )(
+  val viewViaApply: HtmlFormat.Appendable
+  val viewViaRender: HtmlFormat.Appendable
+  val viewViaF: HtmlFormat.Appendable
+
+  def normalPage(messageKeyPrefix: String, messageKeySuffix: String, messageHeadingArgs: Any*)(
     expectedGuidanceKeys: String*
   ): Unit =
     "behave like a normal page" - {
-      val suffix = if (messageKeySuffix.isEmpty) "" else s".$messageKeySuffix"
+      Seq(".apply", ".render", ".f").foreach {
+        case byMethod @ ".apply"  =>
+          pageByMethod(viewViaApply, messageKeyPrefix, messageKeySuffix, messageHeadingArgs: _*)(byMethod)
+        case byMethod @ ".render" =>
+          pageByMethod(viewViaRender, messageKeyPrefix, messageKeySuffix, messageHeadingArgs: _*)(byMethod)
+        case byMethod @ ".f"      =>
+          pageByMethod(viewViaF, messageKeyPrefix, messageKeySuffix, messageHeadingArgs: _*)(byMethod)
+      }
 
-      "rendered" - {
-        "have the correct banner title" in {
-          val doc         = asDocument(view)
-          val bannerTitle = doc.getElementsByClass("hmrc-header__service-name")
-          bannerTitle.text mustBe messages(applicationBuilder().build())("service.name")
-        }
+      "have the correct banner title" in {
+        val doc         = asDocument(viewViaApply)
+        val bannerTitle = doc.getElementsByClass("hmrc-header__service-name")
+        bannerTitle.text mustBe messages(app)("service.name")
+      }
 
-        "display the correct browser title" in {
-          val doc = asDocument(view)
-          assertEqualsMessage(doc, "title", s"$messageKeyPrefix.title$suffix")
-        }
+      "display the correct guidance" in {
+        val doc = asDocument(viewViaApply)
+        for (key <- expectedGuidanceKeys)
+          assertContainsText(doc, messages(app)(s"$messageKeyPrefix.$key"))
+      }
 
-        "display the correct page title" in {
-          val doc = asDocument(view)
-          assertPageTitleEqualsMessage(doc, s"$messageKeyPrefix.heading$suffix", messageHeadingArgs: _*)
+      "contain a timeout dialog" in {
+        val timeoutElm = asDocument(viewViaApply).select("meta[name=\"hmrc-timeout-dialog\"]")
+        if (expectTimeoutDialog) {
+          assert(timeoutElm.size == 1)
+        } else {
+          assert(timeoutElm.size == 0)
         }
```

Comment:
Not sure the reason for this if else. As I understand, the views use the [Layout](https://github.com/hmrc/advance-valuation-rulings-frontend/blob/main/app/views/templates/Layout.scala.html#L55-L65) which has it in place which means all views have this so not sure which scenario it would be false.

Reply:
Discussed to remove as copied previously - should I remove the copied expectedGuidanceKeys test as its not appropriate for this service?

Reply:
Please remove all that isn't needed, we should add stuff when needed and not leave unused code waiting on when it will be used, thanks